pyairtable/api/abstract.py

```diff
@@ -9,22 +9,24 @@
 
 from .params import to_params_dict
 
+MAX_RECORDS_PER_REQUEST = 10
+
 
 class ApiAbstract(metaclass=abc.ABCMeta):
     VERSION = "v0"
     API_BASE_URL = "https://api.airtable.com/"
     API_LIMIT = 1.0 / 5  # 5 per second
     API_URL = posixpath.join(API_BASE_URL, VERSION)
-    MAX_RECORDS_PER_REQUEST = 10
 
     session: requests.Session
     tiemout: Optional[Tuple[int, int]]
 
-    def __init__(self, api_key: str, timeout=None):
+    def __init__(self, api_key: str, max_records_per_request: int, timeout=None):
```

Comment:
I'm on mobile, so that's why I just send random notes here and there.

Is it possible that someone uses those arguments in a positional, rather than keyword, manner?

Then, after an upgrade, some users will have their timeout interpreted as max_records_per_request?

Reply:
timeout is a named keyword argument, and can't be used without passing the name. For better bw compat though, I think i should move it to the end

Reply:
are you sure?

```python
>>> def foo(a: int = 1, c: int = 3) -> None:
...     print(f"{a=}")
...     print(f"{c=}")
... 
>>> def bar(a: int = 1, b: int = 2, c: int = 3) -> None:
...     print(f"{a=}")
...     print(f"{b=}")
...     print(f"{c=}")
... 
>>> foo(101, 102)
a=101
c=102
>>> bar(101, 102)
a=101
b=102
c=3
```

Reply:
This is just a runtime example of course, but not everyone uses static analysis tools in Python. Either way, I think developers are not required to name the arguments - even with strict type checking.

Reply:
Yeah, there isn't another way I think. @gtalarico – thoughts?

Reply:
Agree it should go last for compat.

I forgot what this is called, but maybe we use the `*` separator to require named arg? 
I think that's probably best practice for a lib. 
Maybe we should required named args for all kwargs...?

```
>>> def test(x, y="b", *, z="c"):
...  print(x)
...  print(y)
...  print(z)
...
>>> test(1,2)
1
2
c
>>> test(1, 2, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: test() takes from 1 to 2 positional arguments but 3 were given
>>> test(1, 2, z=3)
1
2
3
```

Reply:
Either way, for now this should be probably be a kwarg not positional. 
#1 because it seems like it should be an optional parameter with a sane default, but #2 to not break existing code on upgrade.
I think I would probably do something like this 👇 ?
```
def __init__(self, api_key: str, timeout: Optional[int]=None, max_records_per_request: Optional[int]):
   max_records_per_request = max_records_per_request or self.DEFAULT_MAX_RECORDS_PER_REQUEST
   ...
```